ttexalens/hardware/baby_risc_debug.py

```diff
@@ -712,13 +712,67 @@ def read_memory(self, address: int) -> int:
         assert self.debug_hardware is not None, "Debug hardware is not initialized"
         return self.debug_hardware.read_memory(address)
 
-    def write_memory(self, address: int, value: int):
+    def read_memory_bytes(self, address: int, size_bytes: int) -> bytes:
+        if self.enable_asserts:
+            self.assert_not_in_reset()
+        self.assert_debug_hardware()
+        assert self.debug_hardware is not None, "Debug hardware is not initialized"
+
+        word_size = 4
+        aligned_start = address - (address % word_size)
+        aligned_end = ((address + size_bytes + word_size - 1) // word_size) * word_size
+
+        words: list[int] = []
+        words_to_read = (aligned_end - aligned_start) // word_size
+        for offset in range(words_to_read):
+            new_addr = aligned_start + offset * word_size
+            word: int = self.debug_hardware.read_memory(new_addr)
+            words.append(word)
+
+        result = b"".join(word.to_bytes(word_size, byteorder="little") for word in words)
```

Comment:
Maybe it is better to use `bytesarray` instead of two lists:

```suggestion
        result = bytearray()
        words_to_read = (aligned_end - aligned_start) // word_size
        for offset in range(words_to_read):
            new_addr = aligned_start + offset * word_size
            word: int = self.debug_hardware.read_memory(new_addr)
            result.extend(word.to_bytes(4, byteorder="little"))
```